### PR TITLE
chore(masthead): update v2 documentation

### LINKED ISSUE #11210: [web-components]: Masthead v2 documentation is outdated
### Description

We didn't sufficiently update the Storybook documentation after all the updates to Masthead v2.

https://carbon-design-system.github.io/carbon-for-ibm-dotcom/v2/web-components/?path=/docs/components-masthead--default

Not an exhaustive list, but here are some things that need updating:
- Script src endpoint is outdated
- New data endpoint is outdated
- Custom navigation instructions are misleading & incomplete
   - Should demonstrate expected shape of data, point to examples, and make note of TypeScript types in services-store package
- Custom search with Typeahead API docs are outdated
   - Use old search endpoint
   - Custom fetch function's response shape differs from what's described (should be `data.response`, not `data.hints`)
- Grouped search has similar issues 

### Component(s) impacted

c4d-masthead family of components

### Browser

_No response_

### Carbon for IBM.com version

v2.0.0-rc2

### Severity

Severity 2 = Aspects of design is broken, and impedes users in a significant way, but there is a way to complete their tasks. Affects major functionality, has a workaround.

### Application/website

IBM.com

### Package

@carbon/ibmdotcom-web-components

### CodeSandbox example

https://carbon-design-system.github.io/carbon-for-ibm-dotcom/v2/web-components/?path=/docs/components-masthead--default

### Steps to reproduce the issue (if applicable)

_No response_

### Release date (if applicable)

_No response_

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/blob/main/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues) for duplicate issues

### PR DESCRIPTION
**Pull request by jkaeser**
### Related Ticket(s)

Resolves #11210

### Changelog

**Changed**

- Updates masthead documentation to match expected behavior in v2.
- Fixes incorrect React wrapper Masthead demonstrations in Storybook.

**Removed**

- Removes docs specific to IBM Cloud that have been made unnecessary in v2.

### CODE REVIEW

File: packages/web-components/src/components/masthead/__stories__/masthead.stories.react.tsx [hunk 1]
```diff
@@ -54,17 +54,24 @@ const scopeParameters = [
...
     .then((response) => response.json())
     .then((data) => {
-      const searchResults = [
-        data.hints,
+      let searchResults = [
+        // Results not including "carbon"
+        data.response
+          .filter(result => !result[0].toLowerCase().includes('carbon'))
+          .map(result => result[0]),
+        // optional grouped category results including "carbon"
```
Comment (m4olivei) [inline packages/web-components/src/components/masthead/__stories__/masthead.stories.react.tsx:66]:
```suggestion
        // Optional grouped category results including "carbon"
```

File: packages/web-components/src/components/masthead/__stories__/README.stories.react.mdx [hunk 2]
```diff
@@ -87,82 +78,120 @@ const urlObject = {
...
+  );
 }
 ```
 
-To set a custom endpoint for the translation service to fetch from, set either
-the `data-endpoint` attribute like below or `C4D_TRANSLATION_ENDPOINT`
-environment variable.
+### Using L1 nav
+
+To use L1 nav, set `l1Data` property. It property should be a MastheadL1 object, as defined in [the services-store package](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/blob/main/packages/services-store/src/types/translateAPI.ts).
```
Comment (m4olivei) [inline packages/web-components/src/components/masthead/__stories__/README.stories.react.mdx:144]:
```suggestion
To use L1 nav, set `l1Data` property. It should be a `MastheadL1` object, as defined in [the services-store package](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/blob/main/packages/services-store/src/types/translateAPI.ts).
```

File: packages/web-components/src/components/masthead/__stories__/README.stories.react.mdx [hunk 3]
```diff
@@ -215,31 +246,36 @@ suggestions in `items`. In the example below, the section `Product pages` is
...
     .then((data) => {
       let searchResults = [
-        data.hints,
-
-        // optional category results fetched from API
+        // Results not including "carbon"
+        data.response
+          .filter(result => !result[0].toLowerCase().includes('carbon'))
+          .map(result => result[0]),
+        // optional grouped category results including "carbon"
```
Comment (m4olivei) [inline packages/web-components/src/components/masthead/__stories__/README.stories.react.mdx:258]:
```suggestion
        // Optional grouped category results including "carbon"
```

File: packages/web-components/src/components/masthead/__stories__/README.stories.react.mdx [hunk 4]
```diff
@@ -248,19 +284,45 @@ The API results must match the following structure:
   ['result 1', 'result 2', 'result 3'],
```
Comment (m4olivei) [inline packages/web-components/src/components/masthead/__stories__/README.stories.react.mdx:284]: Same comment here as above, do we have a type for these API results that we could point people to in the preceeding sentance?

**Review by m4olivei — APPROVED** [review]
This is great. Nice to have this cleaned up. Only tiny suggestions here. Feel free to take or leave and move on. Thanks!

File: packages/web-components/src/components/masthead/__stories__/README.stories.mdx [hunk 5]
```diff
@@ -298,19 +293,45 @@ The API results must match the following structure:
   ['result 1', 'result 2', 'result 3'],
```
[the remark this answers is not in the document]
Comment (jkaeser) [reply packages/web-components/src/components/masthead/__stories__/README.stories.mdx:293]: Unfortunately we don't have any types for this. The shape described here is based on what's returned by [`SearchTypeaheadAPI.getResults()`](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/blob/main/packages/services/src/services/SearchTypeahead/SearchTypeahead.js#L45), which ultimately pings another service that lives at `https://www-api.ibm.com/search/typeahead/v1/`. I don't know who owns that.

That class (and in fact the whole Services package)  is written in vanilla JS, but I suppose we could add type definitions in the Services Store package, or skip all that and just document a type in the Search With Typeahead component.

**Review by kennylam — APPROVED** [review]

**Pull request merged**